From 74c6426e385fc04a6dc9940883a4494bdad1e4f7 Mon Sep 17 00:00:00 2001
From: Manish Dharanenthiran <quic_mdharane@quicinc.com>
Date: Wed, 15 Mar 2023 18:12:30 +0530
Subject: [PATCH] sigma_dut: Disable MU BAR in DL MU MIMO case

For some WFA stations, if MU BAR is enabled in DL MU MIMO
case, PER is increased hence the MCS gets reduced. Because
of this, most of the packets are sent in SU PPDU instead of
MU PPDU.

Fix this by disabling MU_BAR for all DL MU MIMO case.

Signed-off-by: Manish Dharanenthiran <quic_mdharane@quicinc.com>
---
 ap.c | 6 ++++++
 1 file changed, 6 insertions(+)

--- a/ap.c
+++ b/ap.c
@@ -8171,6 +8171,20 @@ static void fwtest_set_he_params(struct
 		fwtest_cmd_wrapper(dut, "-m 0x4b -v 0 21 0", ifname);
 	}
 
+	if (dut->ap_band == BAND_2G && dut->ap_he_mimo == MIMO_DL) {
+	       mac80211_set_mu_brp_commands(dut, ifname);
+	       /* Increase MU traffic */
+	       run_system_wrapper(dut,
+	       		   "ath11k-fwtest -i %s -m 0x48 -v 0 100 2",
+	       		   ifname);
+	}
+
+	/* Disable MU_BAR for DL MU MIMO */
+	if (dut->ap_he_ppdu == PPDU_MU && dut->ap_he_mimo == MIMO_DL)
+		run_system_wrapper(dut,
+				   "ath11k-fwtest -i %s -m 0x47 -v 2 166 1",
+				   ifname);
+
 }
 
 
@@ -9449,13 +9463,6 @@ skip_key_mgmt:
 					fprintf(f, "fils_discovery_max_interval=20\n");
 			}
 		}
-		 if (dut->ap_he_mimo == MIMO_DL) {
-			mac80211_set_mu_brp_commands(dut, ifname);
-			/* Increase MU traffic */
-			run_system_wrapper(dut,
-					   "ath11k-fwtest -i %s -m 0x48 -v 0 100 2",
-					   ifname);
-		}
 
 		find_ap_ampdu_exp_and_max_mpdu_len(dut);
 
